src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCF.java

```diff
@@ -303,221 +525,375 @@ else if (genotype.isCalled() && genotype.isHomRef()) {
+        //for hom-ref variants, drop other ALTs and subset PLs, GQ is recalculated (may be > 0)
+        } else {
+            if (genotype.hasExtendedAttribute(posteriorsKey)) {
+                subsetPosteriorsToRefVersusNonRef(result, gb);
+            } else {
+                final List<Allele> bestAlleles = AlleleSubsettingUtils.calculateMostLikelyAllelesForMonomorphicSite(result, PLOIDY_TWO, 1);
+                final Allele bestAlt = bestAlleles.stream().filter(a -> !a.isReference() && !a.isNonRefAllele()).findFirst().orElse(Allele.NON_REF_ALLELE);  //allow span dels
+                //here we're assuming that an alt that isn't <NON_REF> will have higher likelihood than non-ref, which should be true
+                final GenotypesContext context = AlleleSubsettingUtils.subsetAlleles(result.getGenotypes(),
+                        genotype.getPloidy(), result.getAlleles(), Arrays.asList(result.getReference(), bestAlt),
+                        null, GenotypeAssignmentMethod.BEST_MATCH_TO_ORIGINAL, result.getAttributeAsInt(VCFConstants.DEPTH_KEY, 0));  //BEST_MATCH to avoid no-calling low qual genotypes
+                final Genotype subsetG = context.get(0);
+                gb = new GenotypeBuilder(subsetG).noAttributes();  //remove attributes because hom ref blocks shouldn't have posteriors
+                //subsetting may strip GQ and PLs for low qual genotypes
+                if (!subsetG.hasGQ()) {
+                    gb.GQ(0);
+                }
+                if (!subsetG.hasPL()) {
+                    gb.PL(new int[GenotypeLikelihoods.numLikelihoods(2, genotype.getPloidy())]);  //2 alleles for ref and non-ref
+                }
+            }
+        }
+        if (result.hasAttribute(VCFConstants.DEPTH_KEY)) {
+            final int depth = result.getAttributeAsInt(VCFConstants.DEPTH_KEY, 0);
+            gb.DP(depth);
+            gb.attribute(GATKVCFConstants.MIN_DP_FORMAT_KEY, depth);
+        } else if (genotype.hasAD()) {
+            final int depth = (int) MathUtils.sum(genotype.getAD());
+            gb.DP(depth);
+            gb.attribute(GATKVCFConstants.MIN_DP_FORMAT_KEY, depth);
         }
-        //if GT is not homRef, correct it
-        if (!isHomRefCall(result)) {
-            gb.PL(new int[3]);  //3 for diploid PLs, automatically initializes to zero
-            gb.GQ(0).noAD().alleles(Collections.nCopies(PLOIDY_TWO, newRef)).noAttributes();
+        //NOTE: If we're dropping a deletion allele, then we need to trim the reference and add an END tag with the vc stop position
+        if (result.getReference().length() > 1 || genotype.getAlleles().contains(Allele.SPAN_DEL) || genotype.getAlleles().contains(Allele.NO_CALL)) {
+            newRef = Allele.create(newRef.getBases()[0], true);
         }
+        attrMap.put(VCFConstants.END_KEY, result.getEnd());
+        gb.alleles(Collections.nCopies(genotype.getPloidy(), newRef));
```

Comment:
I'm trying to get rid of no-calls because they cause too many corner cases.  HaplotypeCaller GVCF mode doesn't output no-calls so we won't here either.